src/textual/widgets/_tabs.py

```diff
@@ -0,0 +1,440 @@
+from __future__ import annotations
+
+from typing import ClassVar
+
+import rich.repr
+from rich.style import Style
+from rich.text import Text, TextType
+
+from .. import events
+from ..app import ComposeResult, RenderResult
+from ..binding import Binding, BindingType
+from ..containers import Container, Horizontal, Vertical
+from ..css.query import NoMatches
+from ..geometry import Offset
+from ..message import Message
+from ..reactive import reactive
+from ..renderables.underline_bar import UnderlineBar
+from ..widget import Widget
+from ..widgets import Static
+
+
+class Underline(Widget):
+    """The animated underline beneath tabs."""
+
+    DEFAULT_CSS = """
+    Underline {
+        width: 1fr;
+        height: 1;
+    }
+    Underline > .underline--bar {
+        background: $foreground 10%;
+        color: $accent;
+    }
+    """
+
+    COMPONENT_CLASSES = {"underline--bar"}
+
+    highlight_start = reactive(0)
+    highlight_end = reactive(0)
+
+    class Clicked(Message):
+        """Inform ancestors the underline was clicked."""
+
+        def __init__(self, offset: Offset) -> None:
+            self.offset = offset
+            super().__init__()
+
+    @property
+    def _highlight_range(self) -> tuple[int, int]:
+        """Highlighted range for underline bar."""
+        return (self.highlight_start, self.highlight_end)
+
+    def render(self) -> RenderResult:
+        """Render the bar."""
+        bar_style = self.get_component_rich_style("underline--bar")
+        return UnderlineBar(
+            highlight_range=self._highlight_range,
+            highlight_style=Style.from_color(bar_style.color),
+            background_style=Style.from_color(bar_style.bgcolor),
+        )
+
+    def on_click(self, event: events.Click):
+        """Catch clicks, so that the underline can activate the tabs."""
+        self.post_message(self.Clicked(event.screen_offset))
```

Comment:
Did you forget to stop `event` or do you think it is beneficial to let the original click bubble up? If you think so, can you help me understand why, please?

Reply:
I was thinking that click events don't bubble, but of course they do.